src/qz/printer/info/NativePrinter.java

```diff
@@ -112,6 +118,10 @@ public PrinterProperty<String> getDriver() {
     }
 
     public String getName() {
+        return cachedName.get();
+    }
+
+    private String getNameNative() {
```

Comment:
I don't think we should duplicate our functions just to satisfy a cache.

Perhaps... 

```java
    // PrintService.getName() is slow, use a cache instead per JDK-XXXXXXX
    // TODO: Remove this comment when upstream bug report is filed
    private CachedObject<String> cachedName = new CachedObject<>(printService.value::getName, lifespan);

    public String getName() {
        if (printService != null && printService.value() != null) {
            return cachedName.get();
        }
        return null;
    }

// REMOVE
//    private String getNameNative() {
//    ...    
//    }
```

Reply:
In this instance, calling cachedName.get() would invoke getName(), causing an infinite loop. getNameNative is already private (in retrospect, it's interesting that we have access to the function from CachedObject).

Reply:
No, it's not infinite, per native caller passed to the instantiation:

```
printService.value::getName
```